### ISSUE TITLE
Problem with dependencies - io.searchbox:jest

### ISSUE THREAD

**Issue by cuneytcalishkan** [rfoltyns/log4j2-elasticsearch#30]
Hello,

I think there is a problem with the dependencies towards io.searchbox:jest:2.4.0. This dependency is now forwarded to resolve to version 6.3.1 and there seems to be a change in the API. 
BufferedBulk.getURI() doesn't exist anymore, it takes a parameter of type ElasticsearchVersion
https://github.com/rfoltyns/log4j2-elasticsearch/blob/40d8c4e4c6c919093cf1de3c42c8476162bda7f5/log4j2-elasticsearch-jest/src/main/java/org/appenders/log4j2/elasticsearch/jest/BufferedJestHttpClient.java#L63.

gradle dependencies:
```
implementation "com.lmax:disruptor:3.4.2"
implementation 'io.netty:netty-buffer:4.1.32.Final'
implementation 'org.appenders.log4j:log4j2-elasticsearch-jest:1.3.5'
```
![image](https://user-images.githubusercontent.com/11455741/70341706-d732bc00-1853-11ea-967e-ab1afbb7ee0e.png)

I took the below configuration from your examples.
log4j2 appender configuration:
```xml
<Elasticsearch name="elasticsearchAsyncBatch">
    <RollingIndexName indexName="log4j2_test_jest" pattern="yyyy-MM-dd-HH" timeZone="Europe/Warsaw" />
    <ThresholdFilter level="INFO" onMatch="ACCEPT"/>
    <JacksonJsonLayout>
        <PooledItemSourceFactory poolName="itemPool" itemSizeInBytes="512" initialPoolSize="10000"
                                 monitored="true" monitorTaskInterval="10000" resizeTimeout="500">
            <UnlimitedResizePolicy resizeFactor="0.6" />
        </PooledItemSourceFactory>
    </JacksonJsonLayout>
    <AsyncBatchDelivery batchSize="10000"
                        deliveryInterval="3000" >
        <IndexTemplate name="test_template_jest" path="classpath:indexTemplate.json" />
        <JestBufferedHttp serverUris="http://localhost:9200"
                          connTimeout="500"
                          readTimeout="30000"
                          maxTotalConnection="40"
                          defaultMaxTotalConnectionPerRoute="8"
                          mappingType="index">
            <PooledItemSourceFactory poolName="batchPool" itemSizeInBytes="5120000" initialPoolSize="3"
                                     monitored="true" monitorTaskInterval="10000" resizeTimeout="500">
                <UnlimitedResizePolicy resizeFactor="0.70" />
            </PooledItemSourceFactory>
        </JestBufferedHttp>
    </AsyncBatchDelivery>
</Elasticsearch>
```

As soon as the logger configuration kicks in, the following error occurs:

```
java.lang.NoSuchMethodError: org.appenders.log4j2.elasticsearch.jest.BufferedBulk.getURI()Ljava/lang/String;
	at org.appenders.log4j2.elasticsearch.jest.BufferedJestHttpClient.prepareRequest(BufferedJestHttpClient.java:63) ~[log4j2-elasticsearch-jest-1.3.5.jar:?]
	at org.appenders.log4j2.elasticsearch.jest.BufferedJestHttpClient.executeAsync(BufferedJestHttpClient.java:47) ~[log4j2-elasticsearch-jest-1.3.5.jar:?]
	at org.appenders.log4j2.elasticsearch.jest.JestHttpObjectFactory$1.apply(JestHttpObjectFactory.java:195) ~[log4j2-elasticsearch-jest-1.3.5.jar:?]
	at org.appenders.log4j2.elasticsearch.jest.JestHttpObjectFactory$1.apply(JestHttpObjectFactory.java:178) ~[log4j2-elasticsearch-jest-1.3.5.jar:?]
	at org.appenders.log4j2.elasticsearch.BulkEmitter.notifyListener(BulkEmitter.java:93) ~[log4j2-elasticsearch-core-1.3.5.jar:?]
	at org.appenders.log4j2.elasticsearch.BulkEmitter.stop(BulkEmitter.java:159) ~[log4j2-elasticsearch-core-1.3.5.jar:?]
	at org.appenders.log4j2.elasticsearch.AsyncBatchDelivery.stop(AsyncBatchDelivery.java:182) ~[log4j2-elasticsearch-core-1.3.5.jar:?]
	at org.appenders.log4j2.elasticsearch.ItemSourceAppender.stop(ItemSourceAppender.java:72) ~[log4j2-elasticsearch-core-1.3.5.jar:?]
	at org.appenders.log4j2.elasticsearch.ElasticsearchAppender.lifecycleStop(ElasticsearchAppender.java:240) ~[log4j2-elasticsearch-core-1.3.5.jar:?]
	at org.appenders.log4j2.elasticsearch.ElasticsearchAppender.stop(ElasticsearchAppender.java:219) ~[log4j2-elasticsearch-core-1.3.5.jar:?]
	at org.apache.logging.log4j.core.config.AbstractConfiguration.stop(AbstractConfiguration.java:398) ~[log4j-core-2.12.1.jar:2.12.1]
	at org.apache.logging.log4j.core.AbstractLifeCycle.stop(AbstractLifeCycle.java:136) ~[log4j-core-2.12.1.jar:2.12.1]
	at org.apache.logging.log4j.core.LoggerContext.setConfiguration(LoggerContext.java:584) ~[log4j-core-2.12.1.jar:2.12.1]
	at org.apache.logging.log4j.core.LoggerContext.reconfigure(LoggerContext.java:651) ~[log4j-core-2.12.1.jar:2.12.1]
	at org.apache.logging.log4j.core.LoggerContext.reconfigure(LoggerContext.java:668) ~[log4j-core-2.12.1.jar:2.12.1]
```

Any help is well appreciated, thank you very much for the good work.

Cheers

**Comment by rfoltyns** [comment]
Thank you for posting this. I've been aware of this API incompatibility for a while now and I've yet to come up with a fix.

Since it only affects BufferedBulk class, returning just `_bulk` should do the trick there. It will sacrifice additional query params from `parameterMap` and `cleanApiParameters`, but I'll add support for them eventually if needed. `indexName` and `typeName` are set per Index action, never globally at Bulk, so resulting URL should be still fine. 

I'll release it in 1.3.6 soon.

**Comment by rfoltyns** [comment]
Fixed in 1.3.6.
